lib/composer.js

```diff
@@ -162,6 +168,102 @@ export default class Composer extends Disposable {
     }
   }
 
+  async getDicy (filePath, options = {}) {
+    const magic = new MagicParser(filePath).parse()
+    if (magic.root) {
+      filePath = path.resolve(path.dirname(filePath), magic.root)
+    }
+
+    let dicy
+
+    if (!options.ignoreCache) {
+      dicy = this.cachedDicy.get(filePath)
+      if (dicy) return dicy
+    }
+
+    dicy = await Dicy.create(filePath, options)
+    if (this.cachedDicy.size === 0) {
+      // This is the first Dicy builder so make sure the user options
+      // are synchronized.
+      dicy.updateOptions(this.getDicyOptions(), true)
+    }
+    this.cachedDicy.set(filePath, dicy)
+
+    dicy.on('log', event => {
+      const nameText = event.name ? `[${event.name}] ` : ''
+      const typeText = event.category ? `${event.category}: ` : ''
+      const message = {
+        type: event.severity,
+        text: `${nameText}${typeText}${event.text.replace('\n', ' ')}`
+      }
+
+      if (event.source) {
+        message.filePath = path.resolve(dicy.rootPath, event.source.file)
+        if (event.source.start) {
+          message.range = [[event.source.start - 1, 0], [(event.source.end || event.source.start) - 1, 65535]]
```

Comment:
Extract these "magic numbers" — 65535 — into a constant? Perhaps `MAX_LINE_LENGTH` or whatever it represents. The value seems a bit arbitrary, perhaps consider `Number.MAX_SAFE_INTEGER` instead, unless there is a specific reason for the 64k limit.